### DISCUSSION TITLE
Does supabase support calling auth methods server-side?

### DISCUSSION THREAD

**Discussion by App-Index** [supabase/supabase#14671] [Questions] [answered]
Currently users can log in client-side using JS client library.

If I want to customise the log in request and responses (e.g. put it server side) - can this client library be used? 
E.g. user logs in on server -> custom post-authentication side effects -> return supabase cookie + other cookies?

Or should I achieve this via redirect URL or is there an auth webhook?

**Comment by j4w8n** [comment]
The new pkce method should help. Talked about in this section:

https://supabase.com/docs/guides/auth/server-side-rendering#understanding-the-authentication-flow

**Comment by versoworks** [comment, marked as answer]
> Currently users can log in client-side using JS client library.
> 
> If I want to customise the log in request and responses (e.g. put it server side) - can this client library be used? E.g. user logs in on server -> custom post-authentication side effects -> return supabase cookie + other cookies?
> 
> Or should I achieve this via redirect URL or is there an auth webhook?

Yes, I've implemented it and it works well. Simply call your preferred authentication method with the Supabase client on the server, then send the resulting `Session` to the client and have the client call `setSession` with the returned `Session`.
